src/Orchard/ContentManagement/InfosetHelper.cs

```diff
@@ -129,5 +132,23 @@ public static class InfosetHelper {
             propertyInfo.SetValue(contentPart.Record, value, null);
             contentPart.Store(name, value, versioned);
         }
+
+        /// <summary>
+        /// Checks the given string and throws an <see cref="ArgumentException"/> if it contains characters that are
+        /// invalid in XML. Otherwise just returns the original string.
+        /// </summary>
+        /// <param name="value">The string to check for invalid XML characters.</param>
+        /// <exception cref="ArgumentException">Thrown if the string contains invalid characters.</exception>
+        /// <returns>The original string if no invalid characters were found.</returns>
+        public static string ThrowIfContainsInvalidXmlCharacter(string value) {
+            var invalidCharacters = value.Intersect(InvalidXmlCharacters);
```

Comment:
The intersect will process the whole document. I assume we could fail on the first occurrence. It's not a hot path though, so it doesn't really matter, not blocking the PR.

Reply:
Yeah, I thought about it. The idea was to give the user a more helpful error message by listing all offending characters instead of just the first one. Now this doesn't help much actually with no proper error message being displayed...